go/vt/sqlparser/token.go

```diff
@@ -746,6 +749,15 @@ func (tkn *Tokenizer) next() {
 	tkn.Position++
 }
 
+// reset clears any internal state
+func (tkn *Tokenizer) reset() {
```

Comment:
Either this function, or Scan will need to iterate through all tokens till the next ';', except if a statement was parsed correctly. Otherwise, one error will end up spewing more.

Same thing should be done for ForceEOF.